misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAccess.kt

```diff
@@ -0,0 +1,16 @@
+package misk.web.metadata.all
+
+/**
+ * Bind to set access for the exposed service metadata.
+ *
+ * ```kotlin
+ * // Give internal scraping services access to all metadata for the service
+ * multibind<AccessAnnotationEntry>().toInstance(
+ *   AccessAnnotationEntry<AllMetadataAccess>(
+ *   services = listOf("internal_security_scraper_service"))
+ * )
+ * ```
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.FUNCTION)
+annotation class AllMetadataAccess
```

Comment:
I think we actually want to use the https://github.com/cashapp/misk/blob/ee14b0c62db94432bac204810b786862df9e0ca9/misk-actions/src/main/kotlin/misk/security/authz/Authenticated.kt#L29 in practice. That's of course a deployment decision. Is it possible to bind AccessAnnotationEntries to that existing behaviour of AllowAnyService ?

Reply:
Could you explain more the @Authenticated vs a custom AccessAnnotation? Why one over the other?

Reply:
I imagine that it'd be preferable to have this locked down solely to an internal crawling service exclusively rather than any service?

Reply:
I don't think it should be used directly in misk, but we should retain the option of using those semantics for the AllMetadataAcceess annotation.